interface/src/lib.rs

```diff
@@ -5,16 +5,67 @@
 mod context;
 pub use context::*;
 
-use blake3::Hasher;
+use blake3::{hash, Hasher};
+use serde::{Deserialize, Serialize};
 
-use std::{collections::BTreeMap, convert::TryFrom, error::Error, fmt};
+use std::{cmp::Ordering, collections::BTreeMap, convert::TryFrom, error::Error, fmt};
 
 pub const ADDRESS_LENGTH: usize = 24;
 pub const BYTES32_LENGTH: usize = 32;
+pub const METHOD_SELECTOR_LENGTH: usize = 4;
 pub type Address = [u8; ADDRESS_LENGTH];
 pub type Balance = u64;
 pub type Bytes32 = [u8; BYTES32_LENGTH];
 pub type Bytes = [u8];
+pub type MethodSelectorBytes = [u8; METHOD_SELECTOR_LENGTH];
+pub const METHOD_SELECTOR_DEFAULT: MethodSelector = MethodSelector {
+  bytes: [0u8; METHOD_SELECTOR_LENGTH],
+};
+
+#[derive(Debug, Clone, Deserialize, Eq, PartialEq, PartialOrd, Serialize)]
+pub struct MethodSelector {
+  bytes: MethodSelectorBytes,
+}
+
+impl MethodSelector {
+  pub fn bytes(&self) -> MethodSelectorBytes {
+    self.bytes
+  }
+}
```

Comment:
AFAICT, this is only used to log the selector in hex. How about instead implementing `std::fmt::Display`?
```suggestion
impl std::fmt::Display for MethodSelector {
  fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
    write!(f, "{}", hex::encode(self.0))
  }
}
```